quesma/quesma/search.go

```diff
@@ -146,6 +146,169 @@ type AsyncQuery struct {
 	startTime        time.Time
 }
 
+type executionPlanResult struct {
+	plan         *model.ExecutionPlan
+	err          error
+	responseBody []byte
+}
+
+func (q *QueryRunner) transformQueries(ctx context.Context, plan *model.ExecutionPlan, table *clickhouse.Table) {
+	var err error
+	plan.Queries, err = q.transformationPipeline.Transform(plan.Queries)
+	if err != nil {
+		logger.ErrorWithCtx(ctx).Msgf("error transforming queries: %v", err)
+	}
+
+	plan.Queries, err = registry.QueryTransformerFor(table.Name, q.cfg, q.schemaRegistry).Transform(plan.Queries)
+	if err != nil {
+		logger.ErrorWithCtx(ctx).Msgf("error transforming queries: %v", err)
+	}
+}
+
+func (q *QueryRunner) checkProperties(plan *model.ExecutionPlan, table *clickhouse.Table, queryTranslator IQueryTranslator, ctx context.Context) ([]byte, error) {
+	queries := plan.Queries
+	if len(queries) > 0 && query_util.IsNonAggregationQuery(queries[0]) {
+		if properties := q.findNonexistingProperties(queries[0], table, queryTranslator); len(properties) > 0 {
+			logger.DebugWithCtx(ctx).Msgf("properties %s not found in table %s", properties, table.Name)
+			if elasticsearch.IsIndexPattern(plan.IndexPattern) {
+				return queryparser.EmptySearchResponse(ctx), nil
+			} else {
+				return nil, fmt.Errorf("properties %s not found in table %s", properties, table.Name)
+			}
+		}
+	}
+	return nil, nil
+}
+
+func (q *QueryRunner) executePlanInternal(ctx context.Context, plan *model.ExecutionPlan, queryTranslator IQueryTranslator, table *clickhouse.Table, doneCh chan AsyncSearchWithError, optAsync *AsyncQuery) {
+	go func() {
+		defer recovery.LogAndHandlePanic(ctx, func(err error) {
+			doneCh <- AsyncSearchWithError{err: err}
+		})
+
+		translatedQueryBody, results, err := q.searchWorker(ctx, plan, table, doneCh, optAsync)
+		if err != nil {
+			doneCh <- AsyncSearchWithError{err: err}
+			return
+		}
+
+		if len(results) == 0 {
+			logger.ErrorWithCtx(ctx).Msgf("no hits, sqls: %v", translatedQueryBody)
+			doneCh <- AsyncSearchWithError{translatedQueryBody: translatedQueryBody, err: errors.New("no hits")}
+			return
+		}
+
+		results, err = q.postProcessResults(table, results)
+		if err != nil {
+			doneCh <- AsyncSearchWithError{translatedQueryBody: translatedQueryBody, err: err}
+		}
+
+		if plan.ResultAdapter != nil {
+			results, err = plan.ResultAdapter.Transform(results)
+			if err != nil {
+				doneCh <- AsyncSearchWithError{translatedQueryBody: translatedQueryBody, err: err}
+			}
+		}
+
+		searchResponse := queryTranslator.MakeSearchResponse(plan.Queries, results)
+
+		doneCh <- AsyncSearchWithError{response: searchResponse, translatedQueryBody: translatedQueryBody, err: err}
+	}()
+}
+
+func (q *QueryRunner) executeAlternativePlan(ctx context.Context, plan *model.ExecutionPlan, queryTranslator IQueryTranslator, table *clickhouse.Table, body types.JSON) ([]byte, error) {
```

Comment:
Use named return arguments:
```suggestion
func (q *QueryRunner) executeAlternativePlan(ctx context.Context, plan *model.ExecutionPlan, queryTranslator IQueryTranslator, table *clickhouse.Table, body types.JSON) (responseBody []byte, err error) {
```